Sources/Xcodeproj/pbxproj().swift

```diff
@@ -539,17 +539,34 @@ func xcodeProject(
             if fileSystem.isFile(clangTarget.moduleMapPath) {
                 moduleMapPath = clangTarget.moduleMapPath
                 isGenerated = false
+
+                includeGroup.addFileReference(path: moduleMapPath.asString, name: moduleMapPath.basename)
+                // Save this modulemap path mapped to module so we can later wire it up for its dependees.
+                modulesToModuleMap[target] = (moduleMapPath, isGenerated)
             } else {
-                // Generate and drop the modulemap inside Xcodeproj folder.
-                let path = xcodeprojPath.appending(components: "GeneratedModuleMap", clangTarget.c99name)
-                var moduleMapGenerator = ModuleMapGenerator(for: clangTarget, fileSystem: fileSystem)
-                try moduleMapGenerator.generateModuleMap(inDir: path)
-                moduleMapPath = path.appending(component: moduleMapFilename)
-                isGenerated = true
+                let umbrellaHeaderName = clangTarget.c99name + ".h"
+                if includeGroup.subitems.contains(where: { $0.path == umbrellaHeaderName }) {
+                    // if umbrellaHeader exists, we can use module.
```

Comment:
I would say something like:
```
// If an umbrella header exists, enable Xcode's builtin module's feature rather than generating
// a custom module map. This increases the compatibility of generated Xcode projects.
```

Reply:
That said...

Why can't we always generate a module.modulemap file, and then set the MODULEMAP_FILE build setting to point at it. That should let Xcode's builtin logic work, but can work universally without needing the `else` branch here.